### ISSUE TITLE
[OJS31] Error generating the path of Custom Page in Navigation Menu

### ISSUE THREAD

**Issue by crosfield** [pkp/pkp-lib#2899]
After creating a Custom Page with a path for example 'latex' on the site instead of 'http://%OJSPATH%/latex' OJS gives the path 'http://%OJSPATH%/navigationMenu/view/latex'. Since such the path (if the user isn't logged in) doesn't exist we have the error "The current user is not permitted to preview."

**Comment by asmecher** [comment]
@crosfield, what specific version of OJS are you using? Is this pre-release code on the `master` branch?

**Comment by crosfield** [comment]
@asmecher Yes, from `master` branch just now

**Comment by asmecher** [comment]
@defstat, could you take a look?

**Comment by crosfield** [comment]
<img width="797" alt="screen shot 2017-10-20 at 01 32 55" src="https://user-images.githubusercontent.com/8432203/31797400-c359b544-b536-11e7-8445-723bbca30e7b.png">
<img width="719" alt="screen shot 2017-10-20 at 01 31 49" src="https://user-images.githubusercontent.com/8432203/31797405-c6bff464-b536-11e7-86b5-bf67ffc6f3f5.png">

**Comment by defstat** [comment]
@asmecher PR: #2903

**Comment by asmecher** [comment]
Thanks, @crosfield and @defstat!

**Comment by defstat** [comment]
@asmecher I did change the path but didn't check the "The current user is not permitted to preview." issue.  I will check that now.

**Comment by crosfield** [comment]
Unfortunately, these changes do not help in solving the problem. The problem occurs when the user is logged out. Here are screenshots from the
<img width="802" alt="screen shot 2017-10-20 at 19 21 06" src="https://user-images.githubusercontent.com/8432203/31831582-af511736-b5cc-11e7-92a1-1232c9737eca.png">
<img width="796" alt="screen shot 2017-10-20 at 19 21 44" src="https://user-images.githubusercontent.com/8432203/31831585-b327dafc-b5cc-11e7-96ae-0ea8c08d96fa.png">

 current branch right now.

**Comment by asmecher** [comment]
The fix I just merged is likely all we'll have time to include in the OJS 3.1 release. I'm going to move this issue over to OJS 3.1.1 as it sounds like there's still some work to be done -- @crosfield and @defstat, can you summarize the remaining work for later?

**Comment by crosfield** [comment]
It seems logical to me if the path on front-end of OJS site for the Navigation Menu Custom Page items will be at the same level as other items (non-custom). For example, in my case, instead of the path 'http://%OJSPATH%/navigationMenu/view/latex', there will be the path 'http://%OJSPATH%/latex'. And the path 'http://%OJSPATH%/navigationMenu/view/latex' will work only as a preview for the logged-in users. By the way the Preview button doesn't work right now. Try clicking the Preview button in the Edit window of the Navigation Menu items.

**Comment by asmecher** [comment]
Thanks, @crosfield! @defstat, if the preview button is something you think can be quickly fixed for 3.1 among your other priorities, then please have a look. If you're already too swamped, we can fix this post-3.1.

**Comment by crosfield** [comment]
@asmecher,  @defstat Thank you for your work! And I think rest of this issue is not urgent and probably can be transferred to 3.1.1

**Comment by defstat** [comment]
@asmecher 

PRs for Custom NMIs path fix (context/path and not context/navigationmenu/view/path (the latest is the current configuration)
pkp-lib: #3284
OJS: pkp/ojs#1795

**Comment by asmecher** [comment]
Just some code clean-up comments, @defstat -- when those are ready, please go ahead and merge.

**Comment by defstat** [comment]
@asmecher reminder

**Comment by asmecher** [comment]
@defstat, do you have privileges to merge this yourself?

**Comment by defstat** [comment]
@asmecher it seems that I have privileges for merging on pkp-lib but not on ojs and omp repositories

**Comment by defstat** [comment]
New PRs
pkp-lib: #3335
ojs: pkp/ojs#1820

**Comment by asmecher** [comment]
> it seems that I have privileges for merging on pkp-lib but not on ojs and omp repositories

Thanks for the heads-up, @defstat -- I've added you to both.

**Comment by defstat** [comment]
@asmecher All merged.

**Comment by asmecher** [comment]
Perfect, thanks!
